src/ast/semantic_analyser.cpp

```diff
@@ -2007,6 +2007,12 @@ void SemanticAnalyser::visit(Cast &cast)
 {
   cast.expr->accept(*this);
 
+  if (cast.expr->type.IsRecordTy())
```

Comment:
This also blocks struct to struct casts, should we block those?

Reply:
I'd say yes, C doesn't allow them either.